pkg/manifest/os.go

```diff
@@ -943,6 +914,96 @@ func usersFirstBootOptions(users []users.User) *osbuild.FirstBootStageOptions {
 	return options
 }
 
+func grubStage(p *OS, pt *disk.PartitionTable, kernelOptions []string) *osbuild.Stage {
+	if p.NoBLS {
+		// BLS entries not supported: use grub2.legacy
+		id := "76a22bf4-f153-4541-b6c7-0332c0dfaeac"
+		product := osbuild.GRUB2Product{
+			Name:    p.OSProduct,
+			Version: p.OSVersion,
+			Nick:    p.OSNick,
+		}
+
+		_, err := rpmmd.GetVerStrFromPackageSpecList(p.packageSpecs, "dracut-config-rescue")
+		hasRescue := err == nil
+		return osbuild.NewGrub2LegacyStage(
+			osbuild.NewGrub2LegacyStageOptions(
+				p.Grub2Config,
+				p.PartitionTable,
+				kernelOptions,
+				p.platform.GetBIOSPlatform(),
+				p.platform.GetUEFIVendor(),
+				osbuild.MakeGrub2MenuEntries(id, p.kernelVer, product, hasRescue),
+			),
+		)
+	} else {
+		options := osbuild.NewGrub2StageOptions(pt,
+			strings.Join(kernelOptions, " "),
+			p.kernelVer,
+			p.platform.GetUEFIVendor() != "",
+			p.platform.GetBIOSPlatform(),
+			p.platform.GetUEFIVendor(), false)
+		if cfg := p.Grub2Config; cfg != nil {
+			// TODO: don't store Grub2Config in OSPipeline, making the overrides unnecessary
+			// grub2.Config.Default is owned and set by `NewGrub2StageOptionsUnified`
+			// and thus we need to preserve it
+			if options.Config != nil {
+				cfg.Default = options.Config.Default
+			}
+
+			options.Config = cfg
+		}
+		if p.KernelOptionsBootloader {
+			options.WriteCmdLine = nil
+			if options.UEFI != nil {
+				options.UEFI.Unified = false
+			}
+		}
+		return osbuild.NewGRUB2Stage(options)
+	}
+}
+
+// ukiBootCSVfile creates a file node for the csv file in the ESP which
+// controls the fallback boot to the UKI.
+// NOTE: This is a temporary workaround. We expect that the kernel-bootcfg
+// command from the python3-virt-firmware package will gain the ability to
+// write these files offline during the RHEL 9.7 / 10.1 development cycle.
+func ukiBootCSVfile(espMountpoint, kernelVer, vendor string) (*fsnode.File, error) {
+	kernelFilename := fmt.Sprintf("ffffffffffffffffffffffffffffffff-%s.efi", kernelVer)
+	data := fmt.Sprintf("shimx64.efi,%s,\\EFI\\Linux\\%s ,UKI bootentry\n", vendor, kernelFilename)
+
+	// TODO: what about other architectures
```

Comment:
Perhaps we could assume (can't find a definite quicky) that `aarch64` -> `BOOTAA64` and `x86` -> `BOOTX64`?

Reply:
I looked around a bit and found it hard to find any good references, documentation, or specifications for the CSV file, so I didn't want to make too many assumptions.  But yes, we should parameterise the architecture here for the filename as well as the shim.

Reply:
The BOOT EFI file names are defined in section 3.5.1.1 of the UEFI spec [version 2.11](https://uefi.org/sites/default/files/resources/UEFI_Spec_Final_2.11.pdf) where the arch names are the 'machine type short name' defined by PE format.